### ISSUE TITLE
Different set of patches for PyTorch pinned and main

### ISSUE THREAD

**Issue by pbchekin** [intel/intel-xpu-backend-for-triton#1928]
The current set of patches cannot be applied to PyTorch main:
```
error: patch failed: c10/xpu/XPUDeviceProp.h:136
error: c10/xpu/XPUDeviceProp.h: patch does not apply
error: patch failed: c10/xpu/XPUFunctions.cpp:89
error: c10/xpu/XPUFunctions.cpp: patch does not apply
error: patch failed: test/test_xpu.py:110
error: test/test_xpu.py: patch does not apply
error: patch failed: torch/csrc/xpu/Module.cpp:230
error: torch/csrc/xpu/Module.cpp: patch does not apply
```

https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/10449616738/job/28933849681

**Comment by anmyachev** [comment]
Should it be fixed by https://github.com/intel/intel-xpu-backend-for-triton/pull/1924?

**Comment by pbchekin** [comment]
> Should it be fixed by #1924?

Either we move forward the pin every time or support different set of patches. The former is simpler when it works.
